docs/pages/api-docs/badge.json

```diff
@@ -30,13 +30,7 @@
     },
     "showZero": { "type": { "name": "bool" } },
     "sx": { "type": { "name": "object" } },
-    "variant": {
-      "type": {
-        "name": "union",
-        "description": "'dot'<br>&#124;&nbsp;'standard'<br>&#124;&nbsp;string"
-      },
-      "default": "'standard'"
-    },
+    "variant": { "type": { "name": "string" }, "default": "'standard'" },
```

Comment:
This doesn't seem intended. We now say `Badge` expects any `string` when it only supports `'dot' | 'standard'`: https://deploy-preview-24407--material-ui.netlify.app/api/badge/

Though it's already not ideal on `next`: https://next--material-ui.netlify.app/api/badge/

It should just say `'dot' | 'standard'`, no?

Reply:
Just to clarify: The goal of this PR doesn't need to be `'dot' | 'standard'`. Keeping the current one is fine.

Reply:
I can take a look later on whether we can improve the generation of the docs regarding these kind of props.

Reply:
> I can take a look later on whether we can improve the generation of the docs regarding these kind of props.

When is this later going to be? We already made it worse when introducing `string` (https://github.com/mui-org/material-ui/pull/22140). Now we're making it worse again. Follow-ups are ok but it's clear to me that they're not working in this instance.

Reply:
To be clear, the expected result in the docs is "string"

Reply:
> >We already made it worse when introducing string (#22140).

>worse better smile

The documentation got worse. `Badge#variant` does not support arbitrary strings which the change in #22140 implied.

Expected:
- `Badge#variant` supports `'dot' | 'standard'`
- `UnstyledBadge#variant` supports `string` 

Actual:
- `Badge#variant` supports `string` (was wrong before with `'dot' | 'standard' | string`)

Reply:
I see now, let me take a look on buildApi and what we have there. I think the string may be coming from the unstyled component.

Reply:
I've fixed the issue in the buildAPI, there are no changes to the ` /api/badge` now. For not including the `| string` part, let's resolve it separately. I think it will require more work, or we will need to whitelist which props should have a custom documentation logic, so we can extract the string option out of those (variant, color, size and all other props for which we want to support custom defined values)